src/kaocha/watch.clj

```diff
@@ -262,28 +263,50 @@ errors as test errors."
               (map io/file))
         (:kaocha/tests config)))
 
-(defn watch! [q watch-paths hawk-opts]
-  (hawk/watch! hawk-opts
+(defmulti watch! :type)
+
+(defmethod watch! :hawk [{:keys [q watch-paths opts]}]
+  (hawk/watch! opts
                [{:paths   watch-paths
                  :handler (fn [ctx event]
                             (when (= (:kind event) :modify)
                               (qput q (:file event))))}]))
 
+(defmethod watch! :beholder [{:keys [q watch-paths]}]
+  (apply beholder/watch
+         (fn [{:keys [type path]}]
+           (when (= type :modify)
+             (qput q path)))
+         (map str watch-paths)))
+
 (defn run* [config finish? q]
-  (let [hawk-opts (::hawk-opts config {})
+  (let [watcher-type (::type config :beholder)
+        watcher-opts (condp = watcher-type
```

Comment:
`case` seems clearer than `condp` with `=`. That's just my preference, though.

Reply:
Thanks, I prefer condp 🙈

Reply:
It's not matter of preference here, case is correct. It's the less powerful construct that gets the job done, and so should be preferred. Case is also faster, it can be highly optimized. Use case if you can.

Reply:
Makes sense and yes you are right in this case (:P) case works better. But I once hit a bug something like this:

```clojure
(defn foo [x] (+ 30 x))

(let [x 5
      y 35]
  (case y
    x "not ok"
    (foo 5)"ok"
    "woops"))
```
Which seemed fine until I realised that case only works on compile time constants and does not evaluate the test cases. Since then I just defaulted to using `condp` everywhere as 90% of the time I am never checking the value directly and almost always have a predicate function.

But yeah `case` makes sense here!

Reply:
There is more weird behavior here:
```
(let [x 5]
  (case 5
    x "ok"
    (foo 5) "wat?"
    nil))
;; => "wat?"
```
I came to the conclusion to just avoid case altogether 🙈